### PR TITLE
[DUOS-2709][risk=no] Delete Dataset Updates

### PR DESCRIPTION
**Pull request by rushtong**
### Addresses
https://broadworkbench.atlassian.net/browse/DUOS-2709

### Summary
Minor refactor for deleting a dataset
* Ensure dataset is deletable before trying to delete it
* Move deletion code to DAO
* Ensure dataset is removed from index

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/test/java/org/broadinstitute/consent/http/service/dao/DatasetServiceDAOTest.java [hunk 1]
```diff
@@ -43,6 +43,22 @@ public void setUp() {
...
   }
 
+  @Test
+  void testDeleteDataset() throws Exception {
+    Dataset dataset = createDataset();
+//    when(datasetDAO.findDatasetById(any())).thenReturn(dataset);
+//    when(datasetDAO.insertDatasetAudit(any())).thenReturn(dataset.getDataSetId());
+//    doNothing().when(datasetDAO).deleteUserAssociationsByDatasetId(any());
+//    doNothing().when(datasetDAO).deleteDatasetsProperties(any());
+//    doNothing().when(datasetDAO).deleteConsentAssociationsByDatasetId(any());
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/consent/http/service/dao/DatasetServiceDAOTest.java]: nit: remove or update these checks?

**Review by fboulnois — CHANGES_REQUESTED** [review]
Overall refactor looks good, one recommendation:

File: src/test/java/org/broadinstitute/consent/http/service/dao/DatasetServiceDAOTest.java (hunk 1 above)
Comment (rushtong) [reply src/test/java/org/broadinstitute/consent/http/service/dao/DatasetServiceDAOTest.java]: Yes, thank you!

**Review by fboulnois — APPROVED** [review]
lgtm 👍

**Review by hams7504 — APPROVED** [review]
👍

**Pull request merged**